src/app/providers/NFT/BoosterContext.tsx

```diff
@@ -0,0 +1,92 @@
+'use client'
+import { AVERAGE_BLOCKTIME, NFT_BOOSTER_DATA_URL } from '@/lib/constants'
+import { axiosInstance } from '@/lib/utils'
+import { useQuery } from '@tanstack/react-query'
+import { createContext, ReactNode, useContext, useMemo } from 'react'
+import { Address } from 'viem'
+import { useAccount } from 'wagmi'
+
+interface HolderRewards {
+  estimatedRBTCRewards: bigint
+  estimatedRIFRewards: bigint
+  boostedRBTCRewards: bigint
+  boostedRIFRewards: bigint
+}
+
+interface Holders {
+  [holderAddress: string]: HolderRewards
+}
+
+interface BoostData {
+  nftContractAddress: Address
+  boostPercentage: bigint
+  calculationBlock: bigint
+  holders: Holders
+}
+interface NFTBoosterContext {
+  boostData?: BoostData
+  isLoading?: boolean
+  error?: Error | null
+  currentBoost?: HolderRewards
+  hasActiveCampaign?: boolean
+}
+
+const BoosterContext = createContext<NFTBoosterContext>({
+  boostData: {} as BoostData,
+})
+
+interface BoosterContextProviderProps {
+  children: ReactNode
+}
+export const BoosterProvider = ({ children }: BoosterContextProviderProps) => {
+  const { address } = useAccount()
+  const { data: boostData, isLoading, error, hasActiveCampaign } = useFetchBoostData()
+
+  const value = useMemo(() => {
+    if (!boostData && !isLoading && !error) {
+      return {}
+    }
+
+    if (!address) {
+      return { boostData, isLoading, error, hasActiveCampaign }
+    }
+
+    const currentBoost = boostData?.holders[address]
+
+    return { data: boostData, isLoading, error, currentBoost, hasActiveCampaign }
+  }, [boostData, address, isLoading, error, hasActiveCampaign])
+  return <BoosterContext.Provider value={value}>{children}</BoosterContext.Provider>
+}
+
+export const useNFTBoosterContext = () => useContext(BoosterContext)
+
+const useFetchBoostData = () => {
+  const {
+    data: latestFile,
+    isLoading: isFileDataLoading,
+    error: fileDataError,
+  } = useQuery<string>({
+    queryFn: () => axiosInstance.get(`${NFT_BOOSTER_DATA_URL}/latest`),
```

Comment:
If we `axiosInstance` we are always adding the `CHAIN_ID` as a query param, it does not cause any issue but I dont see any benefit, maybe we just need to use axios.

Reply:
I'm going to change it to use fetch so that we make use of next.js's data caching system